bin/metrics-vsphere.rb

```diff
@@ -0,0 +1,353 @@
+            "storagePath.maxTotalLatency",
+            "virtualDisk.numberReadAveraged",
+            "virtualDisk.numberWriteAveraged",
+            "virtualDisk.read",
+            "virtualDisk.write",
+            "virtualDisk.totalReadLatency",
+            "virtualDisk.totalWriteLatency",
+            "virtualDisk.readOIO",
+            "virtualDisk.writeOIO",
+            "virtualDisk.readLoadMetric",
+            "virtualDisk.writeLoadMetric",
+            "virtualDisk.readIOSize",
+            "virtualDisk.writeIOSize",
+            "virtualDisk.smallSeeks",
+            "virtualDisk.mediumSeeks",
+            "virtualDisk.largeSeeks",
+            "virtualDisk.readLatencyUS",
+            "virtualDisk.writeLatencyUS",
+            "virtualDisk.vFlashCacheIops",
+            "virtualDisk.vFlashCacheLatency",
+            "virtualDisk.vFlashCacheThroughput",
+            "datastore.numberReadAveraged",
+            "datastore.numberWriteAveraged",
+            "datastore.read",
+            "datastore.write",
+            "datastore.totalReadLatency",
+            "datastore.totalWriteLatency",
+            "datastore.sizeNormalizedDatastoreLatency",
+            "datastore.datastoreIops",
+            "datastore.datastoreReadBytes",
+            "datastore.datastoreWriteBytes",
+            "datastore.datastoreReadIops",
+            "datastore.datastoreWriteIops",
+            "datastore.datastoreNormalReadLatency",
+            "datastore.datastoreNormalWriteLatency",
+            "datastore.datastoreReadOIO",
+            "datastore.datastoreWriteOIO",
+            "datastore.datastoreMaxQueueDepth",
+            "datastore.datastoreReadLoadMetric",
+            "datastore.datastoreWriteLoadMetric",
+            "datastore.maxTotalLatency",
+            "datastore.siocActiveTimePercentage",
+            "datastore.datastoreVMObservedLatency",
+            "power.power",
+            "power.powerCap",
+            "power.energy",
+            "hbr.hbrNumVms",
+            "hbr.hbrNetRx",
+            "hbr.hbrNetTx",
+            "vflashModule.numActiveVMDKs",
+            "vsanDomObj.readIops",
+            "vsanDomObj.readThroughput",
+            "vsanDomObj.readAvgLatency",
+            "vsanDomObj.readMaxLatency",
+            "vsanDomObj.readCacheHitRate",
+            "vsanDomObj.readCongestion",
+            "vsanDomObj.writeIops",
+            "vsanDomObj.writeThroughput",
+            "vsanDomObj.writeAvgLatency",
+            "vsanDomObj.writeMaxLatency",
+            "vsanDomObj.writeCongestion",
+            "vsanDomObj.recoveryWriteIops",
+            "vsanDomObj.recoveryWriteThroughput",
+            "vsanDomObj.recoveryWriteAvgLatency",
+            "vsanDomObj.recoveryWriteMaxLatency",
+            "vsanDomObj.recoveryWriteCongestion",
+            "gpu.utilization",
+            "gpu.mem.used",
+            "gpu.mem.usage",
+            "gpu.temperature",
+            "pmem.available.reservation"]
```

Comment:
Constants in ruby can in fact change unless you tell them otherwise:
```ruby
MATRIX = [
  'foo',
  'bar'
].freeze
```

Another [common approach](https://github.com/bbatsov/ruby-style-guide#percent-literal-braces) is to use:
```ruby
MATRIX = %w(
  foo
  bar
)
```

Depending on the version of rubocop it might want parenthesis `%w()` or square brackets `%w[]` with both of these examples you should freeze the constant array.